src/ProcessSlave.php

```diff
@@ -291,37 +291,49 @@ protected function sendCurrentFiles()
      */
     public function run()
     {
-        $this->loop = Factory::create();
+        if (!$this->loop) {
```

Comment:
Is there any chance a client can be run multiple times or why is this needed?

Reply:
As per the fix, yes.

Perhaps we can clean it up by moving the actual bootstrapping logic out of `run()`, since it's now wrapped in a retry-until-it-works loop.

Reply:
Constructor?

Reply:
I've decided to break down the `run()` method so it's not having to guard the creation of the loop instance, and retry logic is distributed between methods instead of nested.

If by constructor you meant to put the loop instantiation there, I'm not sure we'd want to start the loop before `run()` is called, that would make the slaves behave differently to the process manager. I think this is a nicer solution.